### PR TITLE
Decomp BgJyaBombiwa

### PR DESCRIPTION
**Pull request by louist103**

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c [hunk 1]
```diff
@@ -21,17 +23,134 @@ const ActorInit Bg_Jya_Bombiwa_InitVars = {
     (ActorFunc)BgJyaBombiwa_Update,
     (ActorFunc)BgJyaBombiwa_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombiwa/func_808951A0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombiwa/func_80895228.s")
+static ColliderJntSphItemInit sJntSphItemsInit[1] = {
```
Comment (ethteck) [inline src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c:27]: Does this really need to be an array? I imagine it doesn't need to be

File: src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c [hunk 2]
```diff
@@ -21,17 +23,134 @@ const ActorInit Bg_Jya_Bombiwa_InitVars = {
...
+void BgJyaBombiwa_InitCollider(BgJyaBombiwa* this, GlobalContext* globalCtx) {
+    Actor* actor = &this->dyna.actor;
+
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, actor, &sJntSphInit, &this->colliderItems);
+}
+
+void BgJyaBombiwa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    if ((thisx->params & 0x3F) != 0x29) {
+        osSyncPrintf("\x1b[43;30m");
```
Comment (ethteck) [inline src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c:73]: Please use VT macros here. See other files for examples

File: src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c [hunk 3]
```diff
@@ -21,17 +23,134 @@ const ActorInit Bg_Jya_Bombiwa_InitVars = {
...
+    temp.z = this->dyna.actor.posRot.pos.z;
+    func_80033480(globalCtx, &temp, 100.0f, 0xA, 0x64, 0xA0, 1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombiwa/func_8089539C.s")
+void BgJyaBombiwa_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaBombiwa* this = THIS;
+    if ((this->collider.base.acFlags & 2) != 0) {
+        BgJyaBombiwa_BlowUp(this, globalCtx);
+        Flags_SetSwitch(globalCtx, (thisx->params & 0x3F));
```
Comment (ethteck) [inline src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c:143]: The parentheses aroudn the second arg aren't necessary

File: src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c [hunk 4]
```diff
@@ -21,17 +24,140 @@ const ActorInit Bg_Jya_Bombiwa_InitVars = {
...
+    temp.x = this->dyna.actor.posRot.pos.x;
+    temp.y = this->dyna.actor.posRot.pos.y + 70.0f;
+    temp.z = this->dyna.actor.posRot.pos.z;
+    func_80033480(globalCtx, &temp, 100.0f, 0xA, 0x64, 0xA0, 1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombiwa/func_8089539C.s")
+void BgJyaBombiwa_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaBombiwa* this = THIS;
+    if ((this->collider.base.acFlags & 2) != 0) {
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c:148]: line break after the stack var

File: include/functions.h [hunk 5]
```diff
@@ -261,7 +261,7 @@ void EffectSsFhgFlash_Spawn2(GlobalContext *, Actor *, Vec3f *, s16, u8);
 // ? func_80029DBC(?);
 void func_80029E8C(GlobalContext* globalCtx, Vec3f* burstDepthY, Vec3f* burstDepthX, Vec3f* burstOrigin,
                    s16 gravityInfluence, s16 u0, s16 rotSpeed, s16 burstVel, u8 u1, s16 scale, u8 u2, s16 jitter,
-                   s32 duration, s16 u3, s16 objectId, u32 dList);
+                   s32 duration, s16 u3, s16 objectId, Gfx** dList);
```
Comment (Roman971) [inline include/functions.h:264]:
```suggestion
                   s32 duration, s16 u3, s16 objectId, Gfx* dList);
```

File: src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c [hunk 6]
```diff
@@ -9,7 +10,9 @@ void BgJyaBombiwa_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void BgJyaBombiwa_Update(Actor* thisx, GlobalContext* globalCtx);
 void BgJyaBombiwa_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
+void BgJyaBombiwa_SetupDynaPoly(BgJyaBombiwa* this, GlobalContext* globalCtx, void* arg2, DynaPolyMoveFlag flag);
+void BgJyaBombiwa_InitCollider(BgJyaBombiwa* this, GlobalContext* globalCtx);
+
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c:15]: In general we avoid forward declaring functions unless necessary, which isn't the case here.
```suggestion
```

File: src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c [hunk 7]
```diff
@@ -21,17 +24,141 @@ const ActorInit Bg_Jya_Bombiwa_InitVars = {
...
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_jya_bombiwa.c", 174,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void BgJyaBombiwa_InitCollider(BgJyaBombiwa* this, GlobalContext* globalCtx) {
+    Actor* actor = &this->dyna.actor;
+
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, actor, &sJntSphInit, &this->colliderItems);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c:71]:
```suggestion
    Actor* thisx = &this->dyna.actor;

    Collider_InitJntSph(globalCtx, &this->collider);
    Collider_SetJntSph(globalCtx, &this->collider, thisx, &sJntSphInit, &this->colliderItems);
```

File: src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c [hunk 8]
```diff
@@ -21,17 +24,141 @@ const ActorInit Bg_Jya_Bombiwa_InitVars = {
...
+    temp.z = this->dyna.actor.posRot.pos.z;
+    func_80033480(globalCtx, &temp, 100.0f, 0xA, 0x64, 0xA0, 1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombiwa/BgJyaBombiwa_Destroy.s")
+void BgJyaBombiwa_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaBombiwa* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombiwa/func_8089539C.s")
+    if ((this->collider.base.acFlags & 2) != 0) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c:149]:
```suggestion
    if (this->collider.base.acFlags & 2) {
```

File: src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c [hunk 9]
```diff
@@ -21,17 +24,141 @@ const ActorInit Bg_Jya_Bombiwa_InitVars = {
...
+    Vec3f temp;
+    Vec3f temp2;
+    s16 tempS;
+    s8 rotationSpeed;
+    s8 burstVelocity;
+    s32 i;
+    s16 tScale;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombiwa/BgJyaBombiwa_Init.s")
+    for (i = 0; i != 0x10; i++) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Bombiwa/z_bg_jya_bombiwa.c:111]:
```suggestion
    for (i = 0; i < 16; i++) {
```

**Pull request merged**